examples/basic/main.cpp

```diff
@@ -1,6 +1,8 @@
 #include "idstring.h"
 
 int main(int argc, char **argv) {
+    assert(sizeof(istr_dptr_t) == sizeof(char*));  // The promise
```

Comment:
nit: This should technically be a compile time assert. https://stackoverflow.com/questions/807244/c-compiler-asserts-how-to-implement